sdk/instrumentation/net/http/handler.go

```diff
@@ -45,16 +52,33 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		span.SetAttribute(key, value)
 	}
 
-	span.SetAttribute("http.url", r.URL.String())
+	url := r.URL.String()
+	span.SetAttribute("http.url", url)
+
+	// run filters on URL
+	if h.requestFilter.EvaluateURL(span, url) {
+		w.WriteHeader(http.StatusForbidden)
+		return
+	}
 
+	headers := r.Header
 	// Sets an attribute per each request header.
 	if h.dataCaptureConfig.HttpHeaders.Request.Value {
 		SetAttributesFromHeaders("request", NewHeaderMapAccessor(r.Header), span)
 	}
 
+	// run filters on headers
+	if h.requestFilter.EvaluateHeaders(span, headers) {
+		w.WriteHeader(http.StatusForbidden)
+		return
+	}
+
+	shouldRecordBody := h.dataCaptureConfig.HttpBody.Request.Value && ShouldRecordBodyOfContentType(headerMapAccessor{r.Header})
+	shouldFilterByBody := len(h.requestFilter.BodyEvaluators) > 0
```

Comment:
related to the other comment, I think this check may be un-necessary. If we're capturing the body, we should *always* call `filter.evaluate[Request|Response]Body`, even if it results in a noop.

Reply:
done